parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputSplit.java

```diff
@@ -108,32 +99,46 @@ private static long end(List<BlockMetaData> blocks) {
   }
 
   /**
+   * Builds a {@code ParquetInputSplit} from a mapreduce {@link FileSplit}.
+   *
+   * @param split a mapreduce FileSplit
+   * @return a ParquetInputSplit
+   * @throws IOException
+   */
+  public static ParquetInputSplit from(FileSplit split) throws IOException {
+    return new ParquetInputSplit(split.getPath(),
+        split.getStart(), split.getStart() + split.getLength(),
+        split.getLength(), split.getLocations(), null);
+  }
+
+  /**
+   * Builds a {@code ParquetInputSplit} from a mapred
+   * {@link org.apache.hadoop.mapred.FileSplit}.
+   *
+   * @param split a mapreduce FileSplit
+   * @return a ParquetInputSplit
+   * @throws IOException
+   */
+  public static ParquetInputSplit from(org.apache.hadoop.mapred.FileSplit split) throws IOException {
+    return new ParquetInputSplit(split.getPath(),
```

Comment:
Ideally, I'd like to move Hive over to using this and remove the deprecated constructor. In that case, it would need to be public. I also kept the mapreduce version for similar uses. I'm happy to change it, but I think it would be better to keep them exposed.